src/plugin.ts

```diff
@@ -96,18 +96,24 @@ export class HtmlWebpackSkipAssetsPlugin {
                 if (!matcher) {
                     return false;
                 }
+                
+                if (typeof matcher === 'function') {
+                    const matchesCallback = matcher(a);
+                    return !!(matchesCallback);
+                }
+
                 const assetUrl: string = (a.attributes.src || a.attributes.href) as string;
+                if (assetUrl === undefined) {
+                    return false
+                }

```

Comment:
I think I would prefer that the assetURL be refactored to `(a.attributes.src ?? a.attributes.href ?? '') as string;`

That way we catch nulls and undefined values and can allow a pattern to be configured which specifically decides whether or not to skip assets with no src or href value

Reply:
Thanks for the quick review! That makes sense, I've applied your suggestion in https://github.com/swimmadude66/html-webpack-skip-assets-plugin/pull/7/commits/4f0462d7451734c242cb79fd942ab4384a9db201